### DISCUSSION TITLE
Fractional charge behavior for DFT functionals

### DISCUSSION THREAD

**Discussion by sevkel** [pyscf/pyscf#2685] [Q&A] [answered]
Dear PySCF Developers,

im a undergrad student and currently try to plot the fractional charge behaviour (theoretically linear) for the H atom with several functionals. I really dont know how to handle this in PySCF as im new to this. I would be very thankful if you help me out on how to implement that.

best regards and thanks in advance!

**Comment by sunqm** [comment]
You probably need to run uks method. Also you need to modify the `mf.get_occ` method, in which you can assign any occupancy pattern to the variable `mo_occ`.

**Comment by sevkel** [comment]
Thanks for the answer. To run the Code unredtricted makes sense to me, but why do i have to modify the get_occ method?

**Comment by sevkel** [comment]
Can you maybe help me out with an example to get this to run? Or is it alot to implement?

Thanks again!

**Comment by sunqm** [comment, marked as answer]
The SCF iteration in pyscf supports only the integer occupancy, which is governed by the `get_occ` method. To assign fractional occupancy, you need to at least update get_occ . Other methods may need to be overwritten as well. see the APIs https://github.com/pyscf/pyscf/blob/b556049b8104c9fb4735149fa65665f62443565e/pyscf/scf/hf.py#L52 and the example https://github.com/pyscf/pyscf/blob/master/examples/scf/50-mom-deltaSCF.py
